Fixes #13228 - Update Core Deploy Documentation. (#13322)
eclipse/jetty.project: documentation/jetty/modules/operations-guide/pages/jndi/index.adoc

= JNDI

For each specific xref:deploy/index.adoc[_environment_] there is a specific Jetty module that you must enable in order to use JNDI resources in your web application:

> * For Static and Core web applications, the `plus` module.

Reviewer: ```suggestion
* For Static and Core applications, the `plus` module.
```

> * For Jakarta {ee-all-caps} the correspondent `{ee-all}-plus` module.

Reviewer: ```suggestion
* For Jakarta {ee-all-caps} web applications the correspondent `{ee-all}-plus` module.
```

If you have already enabled an xref:annotations/index.adoc[`annotations`] module suitable for your environment, an appropriate `plus` module will already have been enabled, and you do not need to explicitly enable a `plus` module.

If your JNDI resources need extra libraries, for example database driver libraries, that are not located inside your web application, then you must ensure they are on environment class-path.

> You can enable the xref:modules/standard.adoc#ext[`ext`] module for Core web applications, or the specific xref:modules/standard.adoc#eeN-ext[`{ee-all}-ext`] module for Jakarta web applications, and place your extra libraries respectively in `$JETTY_BASE/lib/ext` or `$JETTY_BASE/lib/{ee-all}/ext` to make them available in the environment class-path.

Reviewer: ```suggestion
You can enable the xref:modules/standard.adoc#ext[`ext`] module for Core applications, or the specific xref:modules/standard.adoc#eeN-ext[`{ee-all}-ext`] module for Jakarta web applications, and place your extra libraries respectively in `$JETTY_BASE/lib/ext` or `$JETTY_BASE/lib/{ee-all}/ext` to make them available in the environment class-path.
```

> You can now define JNDI resources and reference them within your web applications.

Reviewer: ```suggestion
You can now define JNDI resources and reference them within your applications.
```

== Defining resources

> JNDI resources are referenced by web application via JNDI names.

Reviewer: ```suggestion
JNDI resources are referenced by applications via JNDI names.
```

> For Jakarta web applications, the JNDI names must be declared via `env-entry`, `resource-ref` and `resource-env-refs` in `web.xml`, `web-fragment.xml` or Jetty's `override-web.xml`.

Reviewer: I thin k this is confusing putting it this way around. I preferred the original text that describes for Jakarta web applications, you need to declare a resource in a Jetty XML file that is either  _internal_ or _external_  to your webapp. You then reference the names of these resources in `web.xml`, `web-fragment.xml` or `web-override.xml`.

External Jetty XML files are those used to deploy the web application, as defined in xref:deploy/index.adoc[this section].

Internal Jetty XML files are those present inside the web application directory or packaged file, as defined in xref:deploy/index.adoc#jetty-web-xml[this section].

For Jakarta web applications, the file `WEB-INF/jetty-{ee-all}-env.xml` is processed _before_ other internal Jetty XML files such as `WEB-INF/jetty-{ee-all}-web.xml`, so that JNDI resources can be bound and later possibly processed by `WEB-INF/jetty-{ee-all}-web.xml`.

See the section on <<xml,Jetty XML files>> for more information on how to choose in which XML file to place the definitions of your JNDI resources.

The definitions of your JNDI resources are instances of the following types:

<<env,`org.eclipse.jetty.plus.jndi.EnvEntry`>>::
Used for `env-entry` resources in `web.xml`.
<<resource,`org.eclipse.jetty.plus.jndi.Resource`>>::
Used for other type of resources.
<<tx,`org.eclipse.jetty.plus.jndi.Transaction`>>::
<<tx,`org.eclipse.jetty.{ee-all}.plus.jndi.Transaction`>>::

> Used for a transaction resource.

Reviewer: Prefer original text. It is a `JTA` transaction manager.

</Configure>
----

> This would be linked into the web application JNDI namespace via an entry in a `web.xml` like so:

Reviewer: ```suggestion
This would be linked into the web application's JNDI namespace via an entry in a `web.xml` like so:
```

</Configure>
----

> The corresponding entry in `web.xml` to bind the ConnectionFactory into the web application JNDI namespace would be:

Reviewer: ```suggestion
The corresponding entry in `web.xml` to bind the ConnectionFactory into the web application's JNDI namespace would be:
```
